### PR TITLE
Ax 428 add baseball endpoints

### PR DESCRIPTION
**Pull request by mretana1999**
# Description
Implemented API calls with Retrofit library which will replace AthleteAPI.dart (which previously used http library). Added API calls for NFL and MLB separately. Created NFLAthlete and MLBAthlete model classes that will contain the JSON objects retrieved from the API. Finally, I also added API call with MLB athlete's ids.

Fixes Jira Ticket #428

## Type of change
Please delete options that are not relevant.
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
Application was run using chrome browser in debugging mode.

# Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have assigned 2 reviewers to check my work
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/pages/DesktopScout.dart [hunk 1]
```diff
@@ -56,6 +61,12 @@ class _DesktopScoutState extends State<DesktopScout> {
     double sportFilterIconSz = 14;
     double _width = MediaQuery.of(context).size.width;
     double _height = MediaQuery.of(context).size.height;
+    final mlbRepo = MLBRepo(MLBAthleteAPI(Dio()));
+    mlbRepo.getAllPlayers().then((response) =>
+        {print("First MLB player retrieved: ${response.first.name}")});
+    final nflRepo = NFLRepo(NFLAthleteAPI(Dio()));
+    nflRepo.getAllPlayers().then((response) =>
+        {print("First NFL player retrieved: ${response.first.name}")});
```
Comment (ketonkss4) [inline lib/pages/DesktopScout.dart:69]: This is out of scope for this ticket we can handle Refactoring the way we call NFL endpoints later

File: lib/pages/DesktopTrade.dart [hunk 2]
```diff
@@ -34,6 +34,7 @@ class _DesktopTradeState extends State<DesktopTrade> {
   @override
   void initState() {
     super.initState();
+    print("This is token list: ${TokenList.tokenList}");
```
Comment (ketonkss4) [inline lib/pages/DesktopTrade.dart:37]: This looks like a debug print statement we should remove it if its not needed

File: lib/pages/ExplorePage.dart [hunk 3]
```diff
@@ -177,8 +177,16 @@ class _ExplorePageState extends State<ExplorePage> {
...
+                                                      war = [
+                                                        athlete.price,
+                                                        athlete.price,
+                                                        athlete.price
+                                                      ],
+                                                      time = [
+                                                        athlete.timeStamp,
+                                                        athlete.timeStamp,
+                                                        athlete.timeStamp
+                                                      ]
```
Comment (ketonkss4) [inline lib/pages/ExplorePage.dart:189]: Why were these changes made? They dont seem relevant to the ticket

File: lib/pages/LandingPage.dart [hunk 4]
```diff
@@ -56,7 +58,7 @@ class _LandingPageState extends State<LandingPage> {
               margin: EdgeInsets.only(bottom: 130.0),
               child: FutureBuilder<dynamic>(
                 // future: AthleteApi.getAthletesLocally(context),
-                future: AthleteApi.getAthletesFromIdsDict(context),
+                future: NFLRepo(NFLAthleteAPI(Dio())).getPlayersById(),
```
Comment (ketonkss4) [inline lib/pages/LandingPage.dart:61]: This is not the ticket to make this change

File: lib/repositories/MlbRepo.dart [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  static final int joseRamirezId = 10000352;
+  static final int marcusSemienId = 10000908;
+  static final int starlingMarteId = 10001009;
+  static final int bryceHarperId = 10001365;
+  static final int carlosCorreaId = 10001918;
+  static final int aaronJudgeId = 10002087;
+  static final int treaTurnerId = 10002094;
+  static final int juanSotoId = 10006794;
+  static final int fernandoTatisId = 10007217;
+  static final int vladimirGuerreroId = 10007501;
```
Comment (ketonkss4) [inline lib/repositories/MlbRepo.dart:16]: Create a datasource you can pass in to provide these values.

File: lib/repositories/NFLRepo.dart [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  NFLRepo(this._api);
+
+  Future<List<NFLAthlete>> getAllPlayers() async {
+    return _api.getAllPlayers();
+  }
+
+  Future<List<NFLAthlete>> getPlayersById() async {
+    return _api.getPlayersById(athleteIdDict);
+  }
+}
```
Comment (ketonkss4) [inline lib/repositories/NFLRepo.dart:24]: Lets save this refactoring for a NFL specific ticket

File: lib/service/TokenList.dart [hunk 7]
```diff
@@ -15,11 +15,11 @@ class TokenList {
     ...AthleteList.list.map((ath) {
       return APT(ath.name + " Long", ath.name + " Long",
           AssetImage('../assets/images/apt.png'), AXT.idToAddress[ath.id]![1]);
-    }),
+    }).toList(),
```
Comment (ketonkss4) [inline lib/service/TokenList.dart:18]: why are these changes being made on this ticket?

File: lib/service/athleteModels/NFLAthlete.dart [hunk 8]
```diff
@@ -0,0 +1,67 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'NFLAthlete.g.dart';
+
+@JsonSerializable()
+class NFLAthlete {
```
Comment (ketonkss4) [inline lib/service/athleteModels/NFLAthlete.dart:6]: Stash these changes and apply them for a NFL specific refactoring ticket

File: lib/service/athlete_api/NFLAthleteAPI.dart [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+abstract class NFLAthleteAPI {
+  factory NFLAthleteAPI(Dio dio, {String baseUrl}) = _NFLAthleteAPI;
+
+  @GET("/players")
+  Future<List<NFLAthlete>> getAllPlayers();
+
+  @POST("/players")
+  Future<List<NFLAthlete>> getPlayersById(
+      @Body() Map<String, List<int>> idsDict);
+}
```
Comment (ketonkss4) [inline lib/service/athlete_api/NFLAthleteAPI.dart:16]: Stash these changes and introduce them when we have an NFL refactoring ticket

**Review by ketonkss4 — CHANGES_REQUESTED** [review]
This ticket is about adding MLB api endpoints to the project and nothing else. Making NFL related tickets on a ticket about MLB will cause unnecessary confusion. Stash all changes related to refactoring the NFL data calls all except for the Rename as that was relevant to MLB so we could differentiate the two data sources.

**Review by ketonkss4 — APPROVED** [review]

**Review by pearlson16 — APPROVED** [review]

**Pull request merged**